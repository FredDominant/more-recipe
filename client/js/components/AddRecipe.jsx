import React from 'react';
import { connect } from 'react-redux';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
/**
 * @description componet holds form to add recipes
 *
 * @class AddRecipe
 * @extends {React.Component}
 */
class AddRecipe extends React.Component {
/**
 * Creates an instance of AddRecipe.
 * @param {any} props
 * @memberof AddRecipe
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      recipeImage: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof AddRecipe
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onUpload(event) {
    const image = event.target.files[0];
    this.setState({ recipeImage: image });
  }
  /**
 *
 * @returns {null} null
 * @memberof AddRecipe
 */
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
   *
   * @returns {null} null
   * @memberof AddRecipe
   */
  addRecipe() {
    uploadImage(this.state.recipeImage)
      .then((response) => {
        // console.log('image url after axios is', response.data.secure_url);
        this.setState({ recipeImage: response.data.secure_url });
        // console.log('image url after updating state is', this.state.recipeImage);
        // dispatch this to store;
      })
      .catch((error) => {
        console.log(error);
      });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addRecipe();
    }
  }
  /**
   *
   * @returns {HTML} html
   * @memberof AddRecipe
   */
  render() {
    const errors = this.state.errors;
    return (
      <div>
        <div className="modal fade" id="addRecipe" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document">
            <div className="modal-content container">
              <div className="modal-header">
                <h5 className="modal-title"><span id="title"><h3 title="More Recipes and cooking tips">More Recipes</h3></span></h5>
                <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                <div className="container error-body">
                  {errors.name && <div className="alert alert-dismissible alert-danger" role="alert">{errors.name}</div>}
                  {errors.description && <div className="alert alert-dismissible alert-danger" role="alert">{errors.description}</div>}
                  {errors.ingredients && <div className="alert alert-dismissible alert-danger" role="alert">{errors.ingredients}</div>}
                  {errors.directions && <div className="alert alert-dismissible alert-danger" role="alert">{errors.directions}</div>}
                </div>
                <form onSubmit={this.handleSubmit}>
                  <div className="form-group">
                    <label htmlFor="exampleInputEmail1">Recipe Name</label>
                    <input
                      type="text"
                      className="form-control"
                      name="name"
                      id="recipeName"
                      aria-describedby="small-recipe-name"
                      placeholder="Awesome Recipe!"
                      value={this.state.name}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipe-description">Recipe Description:</label>
                    <input
                      type="text"
                      className="form-control"
                      name="description"
                      id="recipe-description"
                      placeholder="Awesome sauce made out of nothing!"
                      value={this.state.description}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Ingredients:</label>
                    <textarea
                      className="form-control"
                      id="cook-directions"
                      name="ingredients"
                      rows="3"
                      placeholder="Onions, tomatoes, grilled chicken, nutmeg"
                      value={this.state.ingredients}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Directions to cook:</label>
                    <textarea
                      className="form-control"
                      name="directions"
                      id="cook-directionss"
                      rows="3"
                      placeholder="dice tomatoes, tear chicken, boil onions, mix with nutmeg"
                      value={this.state.directions}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipeImage">Recipe Image</label>
                    <input
                      type="file"
                      className="form-control-file"
                      name="file"
                      id="recipeImage"
                      onChange={this.onUpload}
                    />
                  </div>
                  <button className="btn btn-primary">Add Recipe</button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({

});
export default connect(mapStateToProps)(AddRecipe);
