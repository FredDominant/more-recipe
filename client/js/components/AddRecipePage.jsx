import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import MDSpinner from 'react-md-spinner';
import toastr from 'toastr';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
import addRecipe from '../actions/addRecipe';
/**
 *
 * @class AddRecipePage
 *
 * @extends {React.Component}
 */
export class AddRecipePage extends React.Component {
/**
 * @description Creates an instance of AddRecipePage.
 *
 * @param {any} props
 *
 * @memberof AddRecipePage
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      isUploading: false,
      recipeImage: '/images/noImage.png',
      errors: {},
      addRecipeError: '',
      uploadImageError: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
    this.handleSelectImage = this.handleSelectImage.bind(this);
  }
  /**
 * @description component lifecycle method
 *
 * @returns {null} null
 *
 * @param {nextProp} nextProp object from store
 *
 * @memberof AddRecipe
 */
  componentWillReceiveProps(nextProp) {
    this.setState({ addRecipeError: nextProp.errorMessage });
  }

  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  onUpload(event) {
    const reader = new FileReader();
    reader.onload = (e) => {
      this.setState({ recipeImage: e.target.result });
    };
    reader.readAsDataURL(event.target.files[0]);
  }
  /**
*
* @returns {null} null

* @memberof AddRecipe
*/
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
 *
 * @returns {null} null
 *
 * @memberof AddRecipe
 */
  addNewRecipe() {
    const { recipeImage, name, description, directions, ingredients } = this.state;
    if (recipeImage !== '/images/noImage.png') {
      this.setState({ isUploading: true });
      return uploadImage(recipeImage)
        .then((response) => {
          const recipeUrl = response.data.secure_url;
          console.log('recipe url is', recipeUrl);
          this.setState({ recipeImage: recipeUrl, isUploading: false });
          this.props.createRecipe(this.state);
        })
        .catch(() => {
          this.setState({ isUploading: false, uploadImageError: 'Could not upload image. Recipe not added' });
          toastr.options = {
            closeButton: true
          };
          toastr.error(this.state.uploadImageError);
        });
    }
    console.log('jumped to this place');
    this.props.createRecipe({ name, directions, ingredients, description });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addNewRecipe();
    }
  }
  /**
   *
   * @memberof AddRecipePage
   *
   * @returns {null} null
   */
  handleSelectImage() {
    $('input[type=file]').click();
  }
  /**
   *
   * @returns {null} null
   *
   * @memberof AddRecipePage
   */
  render() {
    const { errors, recipeImage, name, description, directions, ingredients } = this.state;
    return (
      <div className="container-fluid">
        <div id="add-recipe-form" className="container">
          <form onSubmit={this.handleSubmit}>
            <div className="row">
              <div className="col-sm-4">
                <div
                  className="recipe-image"
                  id="add-recipe-image-container"
                  onClick={this.handleSelectImage}
                  role="presentation"
                >
                  <img
                    className="img-thumbnail"
                    src={recipeImage}
                    alt=""
                    srcSet=""
                    id="add-recipe-image"
                  />
                </div>
                <br />
                <input
                  style={{ display: 'none' }}
                  type="file"
                  name="file"
                  id="file-upload"
                  onChange={this.onUpload}
                />
              </div>
              <div className="col-sm-8">
                <div className="form-group">
                  <label htmlFor="recipeName">Recipe Name</label>
                  <input
                    type="text"
                    id="recipeName"
                    className="form-control"
                    name="name"
                    value={name}
                    onChange={this.onChange}
                  />
                  {errors.name && <small className="form-text text-muted">
                    <span className="error-text"> {errors.name} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDescription">Recipe Description</label>
                  <input
                    type="text"
                    id="recipeDescription"
                    className="form-control"
                    name="description"
                    value={description}
                    onChange={this.onChange}
                  />
                  {errors.description && <small className="form-text text-muted">
                    <span className="error-text"> {errors.description} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDirections">Recipe Directions</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeDirections"
                    className="form-control"
                    name="directions"
                    value={directions}
                    onChange={this.onChange}
                  />
                  {errors.directions && <small className="form-text text-muted">
                    <span className="error-text"> {errors.directions} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeingredients">Ingredients</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeIngredients"
                    className="form-control"
                    placeholder="Comma separated list of ingredients"
                    name="ingredients"
                    value={ingredients}
                    onChange={this.onChange}
                  />
                  {errors.ingredients && <small className="form-text text-muted">
                    <span className="error-text"> {errors.ingredients} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <button className="btn btn-primary">
                    {
                      !this.state.isUploading &&
                      <h6 className="text-center"> Add Recipe </h6>
                    }
                    {
                      this.state.isUploading &&
                      <h6 className="text-center"> Uploading Image... </h6>
                    }
                    {
                      this.props.loading &&
                      <span> <MDSpinner /></span>
                    }
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  errorMessage: state.addRecipe.addRecipeErrorMessage,
  addRecipeSuccess: state.addRecipe.addRecipeSuccess,
  loading: state.isUploading
});

const mapDispatchToProps = dispatch => ({
  createRecipe: recipe => dispatch(addRecipe(recipe))
});

AddRecipePage.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};
AddRecipePage.defaultProps = {
  errorMessage: null,
  addRecipeSuccess: null
};
export default connect(mapStateToProps, mapDispatchToProps)(AddRecipePage);

