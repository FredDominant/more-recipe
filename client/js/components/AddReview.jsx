import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import addReview from '../actions/addReview';
/**
 *
 * @class AddReview
 *
 * @extends {React.Component}
 */
export class AddReview extends React.Component {
/**
 * @description Creates an instance of AddReview.
 *
 * @param {object} props
 *
 * @memberof AddReview
 */
  constructor(props) {
    super(props);
    this.state = {
      content: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  /**
   *
   * @param {object} event
   *
   * @memberof AddReview
   *
   * @returns {null} null
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
   * @returns {null} null
   *
   * @param {object} event
   *
   * @memberof AddReview
   */
  handleSubmit(event) {
    const { content } = this.state;
    const recipeId = this.props.recipeId;
    event.preventDefault();
    if (content.trim().length) {
      this.props.postReview(content, recipeId);
      this.setState({ content: '' });
    }
  }
  /**
   *
   *
   * @memberof AddReview
   *
   * @return {ReactElement} markup
   */
  render() {
    return (
      <div>
        <div id="review-form">
          <div>
            <div>
              <form onSubmit={this.handleSubmit}>
                <input
                  type="text"
                  className="form-control"
                  name="content"
                  placeholder="Add a review..."
                  onChange={this.onChange}
                  value={this.state.content}
                  required
                />
                <br />
                <button className="btn" id="add-review-button">
                  <span><i className="fas fa-comment-alt" /> </span>
                  Post Review
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
AddReview.propTypes = {
  postReview: PropTypes.func.isRequired,
  recipeId: PropTypes.number
};
AddReview.defaultProps = {
  recipeId: 0
};
const mapDispatchToProps = dispatch => ({
  postReview: (review, recipeId) => dispatch(addReview(review, recipeId))
});
export default connect(null, mapDispatchToProps)(AddReview);

