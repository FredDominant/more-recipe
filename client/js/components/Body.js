import React from 'react';
import ReactDOM from 'react-dom';
import { Switch, Route } from 'react-router-dom';

import { Home } from './Home';
import { Profile } from './Profile';
import { Recipe } from './Recipe'
import { UserHome } from './UserHome';

export class Body extends React.Component {
  render() {
    return (
			<Switch>
				<Route path='/' exact component={Home} />
				<Route path='/home' exact component={UserHome} />
				<Route path='/recipe' component={Recipe} />
				<Route path='/profile' component={Profile} />
			</Switch>
    );
  }
}
