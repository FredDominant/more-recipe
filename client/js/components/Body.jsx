import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from './Home';
import Profile from './Profile';
import Recipe from './Recipe';
import UserHome from './UserHome';

const Body = () => (
  <Switch>
    <Route path="/" exact component={Home} />
    <Route path="/home" exact component={UserHome} />
    <Route path="/profile" component={Profile} />
    <Route path="/recipe/:recipeId" component={Recipe} />
    <Route path="/user/:userId" component={Profile} />
    <Route component={Recipe} />
  </Switch>
);
export default Body;
