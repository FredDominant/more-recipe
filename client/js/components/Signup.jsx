import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import signupValidator from '../validation/SignupValidator';
import signupUser from '../actions/signupUser';
/**
 * @description this class creates a signup form
 *
 * @class Signup
 * @extends {React.Component}
 */
class Signup extends React.Component {
/**
 * Creates an instance of Signup.
 * @param {any} props
 * @memberof Signup
 */
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      email: '',
      password: '',
      confirmPassword: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  /**
 * @description handles form change events
 * @returns {null} null
 * @param {any} event
 * @memberof Signup
 */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }
  /**
   *
   * @return {dispatch} react-redux dispatch
   * @param {any} event
   * @memberof Signup
   */
  onSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      const { firstname, lastname, email, password, confirmPassword } = this.state;
      this.props.dispatch(signupUser({ firstname, lastname, email, password, confirmPassword }));
    }
  }
  /**
   *
   * @returns {boolean} boolean
   * @memberof Signup
   */
  isValid() {
    const { isValid, errors } = signupValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
   * @description react render method
   *
   * @returns {component} react component
   * @memberof Signup
   */
  render() {
    const { errors } = this.state;
    return (
      <div>
        <div className="modal fade" id="register" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document">
            <div className="modal-content container">
              <div className="row">
                <div className="col-md-9">
                  <span><h1 className="modal-title" id="signup-title" >More Recipes</h1></span>
                  <span><h6 className="signup-text">Register to view cool awesome recipes, rate and review recipes, and create your own recipes and menus.</h6></span>
                </div>
                <div className="col-md-3">
                  <button type="button" className="close cancel-signup" data-dismiss="modal" aria-label="Close">
                    <span aria-hidden="true">&times;</span>
                  </button>
                </div>
              </div>
              <div className="modal-body">
                <div>
                  <div className="container">
                    {errors.firstname && <div className="alert alert-danger" role="alert">{errors.firstname}</div>}
                    {errors.lastname && <div className="alert alert-danger" role="alert">{errors.lastname}</div>}
                    {errors.email && <div className="alert alert-danger" role="alert">{errors.email}</div>}
                    {errors.password && <div className="alert alert-danger" role="alert">{errors.password}</div>}
                    {errors.confirmPassword && <div className="alert alert-danger" role="alert">{errors.confirmPassword}</div>}
                  </div>
                  <form className="form-group" onSubmit={this.onSubmit}>
                    <div className="container">
                      <div className="input-group">
                        <span className="input-group-addon" id="firstName-addon"><i className="fa fa-user" aria-hidden="true" /></span>
                        <input
                          type="text"
                          value={this.state.firstname}
                          onChange={this.onChange}
                          className="form-control signup-form"
                          placeholder="First Name"
                          aria-label="firstName"
                          aria-describedby="firstName-addon"
                          name="firstname"
                        />
                      </div>
                      <br />
                      <div className="input-group">
                        <span className="input-group-addon" id="lastName-addon"><i className="fa fa-user" aria-hidden="true" /></span>
                        <input
                          type="text"
                          alue={this.state.lastname}
                          onChange={this.onChange}
                          className="form-control signup-form"
                          placeholder="Last Name"
                          aria-label="lastName"
                          aria-describedby="lastName-addon"
                          name="lastname"
                        />
                      </div>
                      <br />
                      <div className="input-group">
                        <span className="input-group-addon" id="email-addon"><i className="fa fa-envelope" aria-hidden="true" /></span>
                        <input
                          type="email"
                          value={this.state.email}
                          onChange={this.onChange}
                          className="form-control signup-form"
                          placeholder="Email "
                          aria-label="email"
                          aria-describedby="email-addon"
                          name="email"
                        />
                      </div>
                      <br />
                      <div className="input-group">
                        <span className="input-group-addon" id="password-addon"><i className="fa fa-key" aria-hidden="true" /></span>
                        <input
                          type="password"
                          value={this.state.password}
                          onChange={this.onChange}
                          className="form-control signup-form"
                          placeholder="Password"
                          a
                          ria-label="password"
                          aria-describedby="password-addon"
                          name="password"
                        />
                      </div>
                      <br />
                      <div className="input-group">
                        <span className="input-group-addon" id="confirmPassword-addon"><i className="fa fa-key" aria-hidden="true" /></span>
                        <input
                          type="password"
                          value={this.state.confirmPassword}
                          onChange={this.onChange}
                          className="form-control signup-form"
                          placeholder="Confirm password"
                          aria-label="confirmPassword"
                          aria-describedby="confirmPassword-addon"
                          name="confirmPassword"
                        />
                      </div>
                      <br />
                      <div className="input-group">
                        <button type="submit" className="form-control btn btn-primary signup-form" id="register-button"> <span className="register-text"> Register </span> </button>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Signup.propTypes = {
  dispatch: PropTypes.func.isRequired
};

export default connect(null)(Signup);
