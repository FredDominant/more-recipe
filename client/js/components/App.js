import React from 'react';
import ReactDOM from 'react-dom';

import { Search } from './Search';
import { Navigation } from './Navigation';
import { Body } from './Body';

 export class App extends React.Component {
  render() {
    return (
      <div>
					<Body />
      </div>
    );
  }
};
