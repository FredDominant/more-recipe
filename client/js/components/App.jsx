import React from 'react';

import Body from './Body';
import Errorboundary from '../components/Errorboundary';
import Navbar from './Navbar';

const App = () => (
  <div>
    <Navbar>
      <Errorboundary >
        <Body />
      </Errorboundary>
    </Navbar>
  </div>
);
export default App;
