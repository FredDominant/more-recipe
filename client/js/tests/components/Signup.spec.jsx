import React from 'react';
import { shallow } from 'enzyme';

import { Signup } from '../../components/Signup';

const props = {
  signup: jest.fn(),
  errorMessage: '',
  authenticated: true,
  fetching: false,
  history: {
    push: () => {}
  }
};
/**
 * @description test for Body component
 *
 * @returns {null} null
 */
function setup() {
  const wrapper = shallow(<Signup {...props} />);
  return {
    wrapper
  };
}

describe('Test for Signup component', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
describe('onChange() should', () => {
  it('be called when form input changes', () => {
    const { wrapper } = setup();
    const event = {
      target: { name: 'firstName', value: 'random name' }
    };
    const inputForm = wrapper.find('#firstName');
    inputForm.simulate('change', event);
    expect(wrapper.instance().state.firstName).toBe('random name');
  });
});

describe('handleSubmit() should', () => {
  it('not work for invalid email', () => {
    const wrapper = shallow(<Signup {...props} />);
    const event = {
      preventDefault: jest.fn()
    };
    const form = wrapper.find('.form-group');
    wrapper.setState({
      email: 'random email',
      firstName: 'firstName',
      password: '123456'
    });
    form.simulate('submit', event);
    expect(wrapper.instance().state.errors).toEqual({
      confirmPassword: 'You need to confirm password',
      email: 'Invalid Email',
      lastName: 'Last Name is required',
      password: "Passwords don't match"
    });
    expect(wrapper.instance().props.signup).not.toBeCalled();
  });

  it('should work if form details are valid', () => {
    const wrapper = shallow(<Signup {...props} />);
    const event = {
      preventDefault: jest.fn()
    };
    const form = wrapper.find('.form-group');
    wrapper.setState({
      email: 'email@email.com',
      firstName: 'firstName',
      lastName: 'lastName',
      password: '123456',
      confirmPassword: '123456'
    });
    form.simulate('submit', event);
    expect(wrapper.instance().state.errors).toEqual({});
    expect(wrapper.instance().props.signup).toBeCalled();
  });
});
