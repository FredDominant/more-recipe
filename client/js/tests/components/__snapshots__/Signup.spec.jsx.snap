// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Signup component should render correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalLabel"
    className="modal fade"
    id="register"
    role="dialog"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content container register-signup"
      >
        <div
          className="row container"
        >
          <div
            className="col-xs-2"
          >
            <button
              aria-label="Close"
              className="close cancel-signup"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
                className="close-signup"
              >
                Ã—
              </span>
            </button>
          </div>
        </div>
        <div
          className="row container"
        >
          <div
            className="col-xs-12 text-center"
          >
            <span>
              <h1
                className="modal-title"
                id="signup-title"
              >
                More Recipes
              </h1>
            </span>
            <span>
              <h6
                className="signup-text text-center"
              >
                Register to view cool awesome recipes, rate and review recipes, and create your own recipes and menus.
              </h6>
            </span>
          </div>
        </div>
        <div
          className="modal-body"
        >
          <div>
            <div
              className="container"
            />
            <form
              className="form-group"
              onSubmit={[Function]}
            >
              <div
                className="container"
              >
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                    id="firstName-addon"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user"
                    />
                  </span>
                  <input
                    aria-describedby="firstName-addon"
                    aria-label="firstName"
                    className="form-control signup-form"
                    id="firstName"
                    name="firstName"
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                    value=""
                  />
                </div>
                <br />
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                    id="lastName-addon"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user"
                    />
                  </span>
                  <input
                    aria-describedby="lastName-addon"
                    aria-label="lastName"
                    className="form-control signup-form"
                    name="lastName"
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                    value=""
                  />
                </div>
                <br />
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                    id="email-addon"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-envelope"
                    />
                  </span>
                  <input
                    aria-describedby="email-addon"
                    aria-label="email"
                    className="form-control signup-form"
                    id="signup-email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email "
                    type="email"
                    value=""
                  />
                </div>
                <br />
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                    id="password-addon"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-key"
                    />
                  </span>
                  <input
                    aria-describedby="password-addon"
                    aria-label="password"
                    className="form-control signup-form"
                    id="signup-password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <br />
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                    id="confirmPassword-addon"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-key"
                    />
                  </span>
                  <input
                    aria-describedby="confirmPassword-addon"
                    aria-label="confirmPassword"
                    className="form-control signup-form"
                    id="signup-confirmPassword"
                    name="confirmPassword"
                    onChange={[Function]}
                    placeholder="Confirm password"
                    type="password"
                    value=""
                  />
                </div>
                <br />
                <div
                  className="input-group"
                >
                  <button
                    className="form-control btn signup-form register-button"
                    disabled={false}
                    id="register-button"
                    type="submit"
                  >
                    <span
                      className="register-text"
                    >
                      Register 
                    </span>
                     
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
